pkg/vpcmodel/groupingSelfLoop.go

```diff
@@ -349,21 +350,21 @@ func listOfUniqueEndpoints(oldGroupingSrcOrDst map[string][]*GroupedConnLine, sr
 			}
 		}
 	}
-	return
+	return listOfEndpoints, conn, connProps
 }
 
 // merges a list of GroupedConnLine to be merged
 // input: <old grouping, grouping src or dst, list of key to be merged>
 // output: new key, new []*GroupedConnLine
-func mergeGivenList(oldGroupingSrcOrDst map[string][]*GroupedConnLine, srcGrouping bool,
-	toMergeKeys []string) (newKey string, newGroupedConnLine []*GroupedConnLine) {
-	epsInNewKey, _ := listOfUniqueEndpoints(oldGroupingSrcOrDst, srcGrouping, toMergeKeys)
-	epsInNewLines, conn := listOfUniqueEndpoints(oldGroupingSrcOrDst, !srcGrouping, toMergeKeys)
+func mergeGivenList(oldGroupingSrcOrDst map[string][]*groupedConnLine, srcGrouping bool,
+	toMergeKeys []string) (newKey string, newGroupedConnLine []*groupedConnLine) {
+	epsInNewKey, _, _ := listOfUniqueEndpoints(oldGroupingSrcOrDst, srcGrouping, toMergeKeys)
+	epsInNewLines, conn, commonPros := listOfUniqueEndpoints(oldGroupingSrcOrDst, !srcGrouping, toMergeKeys)
 	for _, epInLineValue := range epsInNewLines {
 		if srcGrouping {
-			newGroupedConnLine = append(newGroupedConnLine, &GroupedConnLine{epInLineValue, &epsInNewKey, conn})
+			newGroupedConnLine = append(newGroupedConnLine, &groupedConnLine{epInLineValue, &epsInNewKey, commonPros})
 		} else {
-			newGroupedConnLine = append(newGroupedConnLine, &GroupedConnLine{&epsInNewKey, epInLineValue, conn})
+			newGroupedConnLine = append(newGroupedConnLine, &groupedConnLine{&epsInNewKey, epInLineValue, commonPros})
 		}
 	}
 	newKey = getKeyOfGroupConnLines(&epsInNewKey, conn)
```

Comment:
If we keep `connStrKey` then I think it better use that computed key value, instead of duplicating the code that calls the right function to re-compute the key.